File: C#/DetrackSample/Models/DeliveryItem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Newtonsoft.Json;

namespace DetrackSample.Models
{
  /// <summary>
  /// Delivery Item model for Add/Edit Deliveries
  /// Mark properties as required as defined by Detrack API
  /// </summary>
  public class DeliveryItem
  {
    [JsonProperty(Required = Required.Always)]
    public string date { get; set; }

    [JsonProperty(Required = Required.Always, PropertyName = "do")]
    public string Do { get; set; }

    [JsonProperty(Required = Required.Always)]
    public string address { get; set; }
    public string delivery_time { get; set; }
    public string deliver_to { get; set; }
    public string phone { get; set; }
    public string notify_email { get; set; }
    public string notify_url { get; set; }
    public string assign_to { get; set; }
    public string instructions { get; set; }
    public string zone { get; set; }
    public List<Item> items { get; set; }
  }
}

File: Ruby/DetrackAPI/Models/DeliveryItem.rb
class DeliveryItem
  def initialize
    @items = []
  end

  def date
    return @date
  end

  def date=(value)
    @date = value
  end

  def do
    return @do
  end

  def do=(value)
    @do = value
  end

  def address
    return @address
  end

  def address=(value)
    @address = value
  end

  def delivery_time
    return @delivery_time
  end

  def delivery_time=(value)
    @delivery_time = value
  end

  def deliver_to
    @deliver_to
  end

  def deliver_to=(value)
    @deliver_to = value
  end

  def phone
    return @phone
  end

  def phone=(value)
    @phone = value
  end

  def notify_email
    return @notify_email
  end

  def notify_email=(value)
    @notify_email = value
  end

  def notify_url
    return @notify_url
  end

  def notify_url=(value)
    @notify_url = value
  end

  def assign_to
    return @assign_to
  end

  def assign_to=(value)
    @assign_to = value
  end

  def instructions
    return @instructions
  end

  def instructions=(value)
    @instructions = value
  end

  def zone
    return @zone
  end

  def zone=(value)
    @zone = value
  end

  def items
    return @items
  end

  def items=(value)
    @items = value
  end
end